application/controllers/admin/globalsettings.php

```diff
@@ -480,6 +480,65 @@ public function surveysettingmenues()
         Yii::app()->getController()->renderPartial('super/_renderJson', ['data' => $menues[0]]);
     }
 
+    public function sendTestEmail()
+    {
+        $sSiteName = Yii::app()->getConfig('sitename');
+
+        //Use the current user details for the default administrator name and email
+        $user = User::model()->findByPk(Yii::app()->session['loginID']);
+        $sTo = $user->full_name . " <" . $user->email . ">";
+        $sFrom = Yii::app()->getConfig("siteadminname") . " <" . Yii::app()->getConfig("siteadminemail") . ">";
+        $sSubject = sprintf(gT('Test email from %s'), $sSiteName);
+        
+        $body   = array();
+        $body[] = sprintf(gT('This is a test email from %s'), $sSiteName);
+        $body   = implode("\n", $body);
+
+        $this->_sendEmailAndShowResult($body, $sSubject, $sTo, $sFrom);
+    }
+
+    private function _sendEmailAndShowResult($body, $sSubject, $sTo, $sFrom)
+    {
+        $mailer = new \LimeMailer();
+        $mailer->rawBody = $body;
+        $mailer->rawSubject = $sSubject;
+        $mailer->setTo($sTo);
+        $mailer->setFrom($sFrom);
+
+        $success = $mailer->sendMessage();
+        $maildebug = $mailer->getDebug('html');
+
+        if ($success) {
+            $content = gT('Email sent successfully');
+        } else {
+            $content = gT('Email sending failure.');
+        }
+
+        $data = [];
+        $data['message'] = $content;
+        $data['success'] = $success;
+        $data['maildebug'] = $maildebug;
+
+        $this->_renderWrappedTemplate('globalsettings', '_emailTestResults', $data);
+    }
+
+    public function sendTestEmailConfirmation()
+    {
+        $user = User::model()->findByPk(Yii::app()->session['loginID']);
+        $aData = [
+            'testEmail' => $user->email,
+            'siteadminemail' => Yii::app()->getConfig("siteadminemail"),
+            'siteadminname' => Yii::app()->getConfig("siteadminname"),
+            'emailmethod' => Yii::app()->getConfig("emailmethod"),
+            'emailsmtphost' => Yii::app()->getConfig("emailsmtphost"),
+            'emailsmtpuser' => Yii::app()->getConfig("emailsmtpuser"),
+            'emailsmtppassword' => 'somepassword',
+            'emailsmtpssl' => Yii::app()->getConfig("emailsmtpssl"),
+            'emailsmtpdebug' => Yii::app()->getConfig("emailsmtpdebug"),
```

Comment:
Why not force emailsmtpdebug ? Or by a checkbox in the form ?

Reply:
The general setting is being used. Why forcing would be better? or why checbox?

Reply:
Because: if you do a test : you want more information than in real.

If user set `emailsmtpdebug` to 0 : no debug return.